File: fixtures/scale/python/large/cipherlab/checks/check_trithemius.py
from cipherlab.ciphers.trithemius import trithemius_encrypt, trithemius_decrypt
from cipherlab.specs.trithemius_spec import trithemius_spec
from cipherlab.corpus.corpus_additive import corpus_additive


def check_trithemius():
    spec = trithemius_spec()
    for text in corpus_additive():
        sealed = trithemius_encrypt(text, spec["key"])
        opened = trithemius_decrypt(sealed, spec["key"])
        if opened != text:
            return False
    return spec["category"] == "additive"

File: fixtures/scale/ruby/large/lib/check_trithemius.rb
require_relative 'trithemius_encrypt'
require_relative 'trithemius_decrypt'
require_relative 'corpus_additive'

def check_trithemius
  corpus_additive.each do |text|
    sealed = trithemius_encrypt(text, 5)
    opened = trithemius_decrypt(sealed, 5)
    return false if opened != text
  end
  true
end
